src/withPlugins.ts

```diff
@@ -1,9 +1,20 @@
 import { Plugins, DiceRule } from './rules/types';
 import createDiceRoller, { defaultPlugins } from './createDiceRoller';
+import defaultRandom, { Random } from "./util/random"
 
-function withPlugins(...plugins: DiceRule<any>[]) {
+function withPlugins(...plugins: DiceRule<any>[]): ReturnType<typeof createDiceRoller>
+function withPlugins(plugins: DiceRule<any>[], random?: Random): ReturnType<typeof createDiceRoller>
```

Comment:
Something about this overload makes me feel a little uneasy. I bet we could clean this up if removed the `createDiceRoller` from this function and did something like this instead:
```js
const diceRoller = createDiceRoller(
  withPlugins(...myPlugins),
  myRandom
)
```
or maybe `withPlugins` could return a function to create the dice roller
```js
const diceRoller = withPlugins(...myPlugins)(myRandom)
```

This would make it a breaking change, but I don't think that's a huge deal given how few people use this. xD

Reply:
We can just bump and release as version 2 :). ALso more people should use this library. It is the best one from my evaluation that gives you the best control over how you want to display the result :) Kudos on that!